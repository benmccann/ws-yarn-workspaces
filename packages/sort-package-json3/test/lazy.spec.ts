import { sortPackageJson } from '../index';

test(`exports`, () =>
{
	let data = {
		"name": "my-package",
		"type": "module",
		"exports": {
			".": {
				// Entry-point for TypeScript resolution - must occur first!
				"types": "./types/index.d.ts",
				// Entry-point for `import "my-package"` in ESM
				"import": "./esm/index.js",
				// Entry-point for `require("my-package") in CJS
				"require": "./commonjs/index.cjs",
			},
			"./k2": {

				// Entry-point for `require("my-package") in CJS
				"require": "./commonjs/index.cjs",
				// Entry-point for TypeScript resolution - must occur first!
				"types": "./types/index.d.ts",
				// Entry-point for `import "my-package"` in ESM
				"import": "./esm/index.js",
			},
		},
		// CJS fall-back for older versions of Node.js
		"main": "./commonjs/index.cjs",
		// Fall-back for older versions of TypeScript
		"types": "./types/index.d.ts",
	};

	let actual = sortPackageJson(data);

	expect(actual).toMatchSnapshot();
	expect(JSON.stringify(actual, null, 2)).toMatchSnapshot();

});

