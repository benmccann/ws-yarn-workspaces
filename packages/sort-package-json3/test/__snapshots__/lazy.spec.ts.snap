// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports 1`] = `
{
  "exports": {
    ".": {
      "import": "./esm/index.js",
      "require": "./commonjs/index.cjs",
      "types": "./types/index.d.ts",
    },
    "./k2": {
      "import": "./esm/index.js",
      "require": "./commonjs/index.cjs",
      "types": "./types/index.d.ts",
    },
  },
  "main": "./commonjs/index.cjs",
  "name": "my-package",
  "type": "module",
  "types": "./types/index.d.ts",
}
`;

exports[`exports 2`] = `
"{
  "name": "my-package",
  "type": "module",
  "exports": {
    ".": {
      "types": "./types/index.d.ts",
      "require": "./commonjs/index.cjs",
      "import": "./esm/index.js"
    },
    "./k2": {
      "types": "./types/index.d.ts",
      "require": "./commonjs/index.cjs",
      "import": "./esm/index.js"
    }
  },
  "main": "./commonjs/index.cjs",
  "types": "./types/index.d.ts"
}"
`;
